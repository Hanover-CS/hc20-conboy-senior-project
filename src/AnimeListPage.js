/* Author: Patrick Conboy
 * Date Created: 1/4/2020
 * Description: This file will be the React component that
 * is the personal anime list webpage for each user.
 */

import React, { Component } from 'react'

class AnimeListPage extends Component {
  render() {
    return (
      <div className="AnimeListPage">
        <h3>Welcome to your personal anime list!</h3>
      </div>
    )
  }
}

export default AnimeListPage;