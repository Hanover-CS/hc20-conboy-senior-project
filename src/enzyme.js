/* Author: Patrick Conboy
 * Date Created: 1/1/2019
 * Description: This file initializes Enzyme for testing purposes.
 */

import Enzyme, { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
export { shallow, mount, render };
export default Enzyme;