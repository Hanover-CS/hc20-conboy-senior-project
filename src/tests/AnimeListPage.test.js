import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { shallow } from '../enzyme'

import AnimeListPage from '../AnimeListPage'

describe('AnimeListPage Tests', () => {
  test('anime list page renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<AnimeListPage />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('anime list page renders correctly', () => {
    const wrapper = shallow(<AnimeListPage />)

    expect(wrapper.find('#AnimeListPage')).toBeDefined()
  })

  test('snapshot of anime list page hasn\'t changed', () => {
    const tree = renderer.create(<AnimeListPage />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})