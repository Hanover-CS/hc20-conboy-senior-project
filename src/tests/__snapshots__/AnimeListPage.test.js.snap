// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimeListPage Tests snapshot of anime list page hasn't changed 1`] = `
<div>
  <div>
    <ul />
  </div>
</div>
`;
